pkg/core/interop/iterator/interop.go

```diff
@@ -36,14 +36,15 @@ func IsIterator(item stackitem.Item) bool {
 	return ok
 }
 
-// Values returns an array of up to `max` iterator values. The second
-// return parameter denotes whether iterator is truncated.
-func Values(item stackitem.Item, max int) ([]stackitem.Item, bool) {
+// Values returns an array of up to `max` iterator values. The provided
+// iterator can safely be reused to retrieve the rest of its values in the
+// subsequent calls to Values.
+func Values(item stackitem.Item, max int) []stackitem.Item {
 	var result []stackitem.Item
 	arr := item.Value().(iterator)
-	for arr.Next() && max > 0 {
+	for max > 0 && arr.Next() {
```

Comment:
Not a useful change.

Reply:
It is a useful change. Imagine that iterator have more elements than `max`. `arr.Next()` receives the subsequent element from the underlying channel and doesn't cache it which makes `arr` unsuitable for further iterations, because we use `arr.Next` one more time at the start of each subsequent request handling. This single element will be lost.